client/rest/src/main/java/org/elasticsearch/client/Response.java

```diff
@@ -22,26 +22,25 @@
 import org.apache.http.Header;
 import org.apache.http.HttpEntity;
 import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
 import org.apache.http.RequestLine;
 import org.apache.http.StatusLine;
 import org.apache.http.client.methods.CloseableHttpResponse;
 
-import java.io.Closeable;
-import java.io.IOException;
 import java.util.Objects;
 
 /**
  * Holds an elasticsearch response. It wraps the {@link CloseableHttpResponse} response and associates it with
  * its corresponding {@link RequestLine} and {@link HttpHost}.
  * It must be closed to free any resource held by it, as well as the corresponding connection in the connection pool.
```

Comment:
it is not a loss... with the async client you have the notion of async response consumers, see `HttpAsyncResponseConsumer`.  That one is in charge of reading the response and closing it when done. No need to close the response on the client side all the time, which is quite a win I think.

The only thing is that we use a default async consumer when not provided, same that async http client does, and that one has a limitation on max Integer.MAX_VALUE content-length, and buffers the response content in memory, see `BasicAsyncResponseConsumer`.